### PR TITLE
Metadata command

### PR DESCRIPTION
**Pull request by noahmacaskill**
Metadata command now produces dummy metadata output for each given file, in the given format. Tests coming.

### CODE REVIEW

**Comment by atruskie** [comment]
Ok, the task we're about to undertake is potentially monolithic. We've got three devs so I want to make sure we can split the work between us.

So let's write a test that essentially confirms one output record is generated per file provided. Then we'll close this PR and move on to making the metadata extractors for the other file formats.

**Comment by noahmacaskill** [comment]
Found that test to be more of a headache than it should have been. I struggled to grab the console output to test there was one record produced per file, so ended up just making a string writer and rerouting the output there, eventually got this to work. I'm using the default ToString output format here, which for some reason is consistently producing one more record than there are files (should note I tested with the JSON serializer as well which didn't produce the extra record), still looking into this. 

It's passing but can probably be much cleaner, feel free to make any suggestions or change it yourself however you see fit.

**Comment by atruskie** [comment]
Actually, you did pretty well.

An in-memory string writer is a great solution.

Note: each formatter can choose to append its own header or footer.  E.g. the JSON formatter emits a `{` as a header and a `}` as a footer.

I'd say the extra new line you're dealing with is the headers / footers for the formatter.


Suggestions:

- Change what you're testing. You're asserting a certain number of new lines exist. I'd rather assert that the strings `/a.WAV`, `/b.WAV`, `/c.WAV` exist in the output.
- Use one of the default formatters (Default, Compact, JSON, JSON-L, CSV). JSON-L for example doesn't print any header or footer so you could easily count (or better yet filter for lines) that start with a `{` and then checks the expected file names have been printed

File: src/MetadataUtility/Commands/EmuCommand.cs [hunk 1]
```diff
@@ -21,6 +21,8 @@ public EmuCommand()
 
             this.AddGlobalOption(FormatOption);
 
+            this.AddGlobalOption(OutOption);
+
```
Comment (atruskie) [inline src/MetadataUtility/Commands/EmuCommand.cs:25]: I don't think we have support for changing the output do we?

In which case exposing this options is a breaking change?

**Review by atruskie — CHANGES_REQUESTED** [review]
Looks great.

I'd merge this in except I'm dubious about exposing the `--output` option. Why was that changed? Does it work? Is there a test asserting it works?

Surprising myself, it looks like I started to add support for this here: https://github.com/QutEcoacoustics/emu/blob/master/src/MetadataUtility/Cli/OutputSink.cs

But it is not finished and needs some work. I'd say revere the change to _EmuCommand_ and do another PR to fix up the out option

**Comment by noahmacaskill** [comment]
The output option actually does seem to work, I did have to change the name to --output though, without that change it wasn't binding to [this].(https://github.com/QutEcoacoustics/emu/blob/7fddbf2dd6da9388dd2180593edb7be6623e047a/src/MetadataUtility/Commands/EmuGlobalOptions.cs#L19 If I'm understanding correctly, System.CommandLine binds options to variables, given the names match up.

That said though, I agree, it makes more sense to do this in another pull request with tests implemented and some more functionality, the changes have been reversed. Something like "-F JSON --output ./out.json" works just fine, but "-F JSON --output ./" throws an error, in the case where just a directory is given, do we want to create a file in that directory with a default name? "-F JSON --output ./out.csv" also works, but spits JSON formatted results to a CSV file, do we want to force the file extension to reflect the output format?

Interested in hearing your thoughts, I could work on this next week, or we can save it for later down the road.

**Comment by noahmacaskill** [comment]
I've squashed the commits down into one again as well.

**Comment by atruskie** [comment]
Yeah, you intuited correctly. The problem was just more the lack of tests and the multi-concerned PR.

I've created #112 to track this issue. I do consider it to be low priority (as one can typically just do `emu -F JSON ... > output.json`) but if it is important to your stakeholders then go for it.

**Review by atruskie — APPROVED** [review]

**Pull request merged**